PhysicsTools/JetMCAlgos/plugins/HadronAndPartonSelector.cc

```diff
@@ -249,12 +251,20 @@ HadronAndPartonSelector::produce(edm::Event& iEvent, const edm::EventSetup& iSet
    }
 
    // select partons
-   if ( partonMode_!="Undefined" )
+   if ( partonMode_!="Undefined" ) {
      partonSelector_->run(particles,partons);
+     for(reco::GenParticleCollection::const_iterator it = particles->begin(); it != particles->end(); ++it)
+     {
+       if( !(it->status()==3 || (( partonMode_=="Pythia8" ) && (it->status()==23)))) continue;
```

Comment:
The problem is that the algorithmic definition cannot be reproduced from MiniAOD since we only keep partons with pT>5 GeV. Maybe for now we should keep them and re-evaluate this at some point in the future.